deploy/buildtools/buildafi.py

```diff
@@ -5,170 +5,240 @@
         message_body += ".\nInspect the log output from IP address " + env.host_string + " for more information."
 
         send_firesim_notification(message_title, message_body)
 
         rootLogger.info(message_title)
         rootLogger.info(message_body)
-        rootLogger.info("Terminating the build instance now.")
-        buildconfig.terminate_build_instance()
 
+        build_config.build_farm_host_dispatcher.release_build_farm_host()
 
-    rootLogger.info("Running process to build AGFI from verilog.")
+    rootLogger.info("Building AWS F1 AGFI from Verilog")
 
-    # First, Produce dcp/tar for design. Runs on remote machines, out of
-    # /home/centos/firesim-build/ """
-    ddir = get_deploy_dir()
-    buildconfig = global_build_config.get_build_by_ip(env.host_string)
-    builddir = buildconfig.get_build_dir_name()
-    # local AWS build directory; might have config-specific changes to fpga flow
-    fpgabuilddir = "hdk/cl/developer_designs/cl_" + buildconfig.get_chisel_triplet()
-    remotefpgabuilddir = "hdk/cl/developer_designs/cl_firesim"
+    local_deploy_dir = get_deploy_dir()
+    fpga_build_postfix = "hdk/cl/developer_designs/cl_{}".format(build_config.get_chisel_triplet())
+    local_results_dir = "{}/results-build/{}".format(local_deploy_dir, build_config.get_build_dir_name())
 
-    # first, copy aws-fpga to the build instance. it will live in
-    # firesim-build/platforms/f1/
-    with StreamLogger('stdout'), StreamLogger('stderr'):
-        run('mkdir -p /home/centos/firesim-build/platforms/f1/')
-    # do the rsync, but ignore any checkpoints that might exist on this machine
-    # (in case builds were run locally)
-    # extra_opts -l preserves symlinks
-    with StreamLogger('stdout'), StreamLogger('stderr'):
-        rsync_cap = rsync_project(local_dir=ddir + "/../platforms/f1/aws-fpga",
-                      remote_dir='/home/centos/firesim-build/platforms/f1/',
-                      ssh_opts="-o StrictHostKeyChecking=no",
-                      exclude=["hdk/cl/developer_designs/cl_*"],
-                      extra_opts="-l", capture=True)
-        rootLogger.debug(rsync_cap)
-        rootLogger.debug(rsync_cap.stderr)
-        rsync_cap = rsync_project(local_dir=ddir + "/../platforms/f1/aws-fpga/{}/*".format(fpgabuilddir),
-                      remote_dir='/home/centos/firesim-build/platforms/f1/aws-fpga/' + remotefpgabuilddir,
-                      exclude=['build/checkpoints'],
-                      ssh_opts="-o StrictHostKeyChecking=no",
-                      extra_opts="-l", capture=True)
+    # cl_dir is the cl_dir that is either local or remote
+    # if locally no need to copy things around (the makefile should have already created a CL_DIR w. the tuple)
+    # if remote (aka not locally) then you need to copy things
+    cl_dir = ""
+    local_cl_dir = "{}/../platforms/f1/aws-fpga/{}".format(local_deploy_dir, fpga_build_postfix)
+
+    # copy over generated RTL into local CL_DIR before remote
```

Comment:
Yes. I'm not sure this is needed IMO. In the past, this was done in the "run replace_rtl" Python function: https://github.com/firesim/firesim/blob/e869ef22f84b631789e792ba76a26e001b38b5df/deploy/buildtools/buildafi.py#L34-L45. I moved it here since I kept the driver build and replace-rtl steps only the `make` commands. I can move this back there if wanted.

Reply:
@timsnyder-siv @sagark Do you have any objections to removing this, and just documenting where these files should exist on a host-to-host basis?

Reply:
This elaborated RTL has always seemed like a duplicate to me.  As long as the cl_dir is being copied back into local_results_dir, I don't care if the duplicate copy of the verilog is being copied back at the top.

Reply:
I agree with Tim here. No matter if Vivado fails or not, build results/logs/etcs are copied back. They should just get the result from the `local_results_dir` (as for documentation, we can just say that the `firesim-generated.sv` file is located in one of the sub-folders of `cl_firesim` - https://docs.fires.im/en/stable/Advanced-Usage/Manager/Manager-Tasks.html?highlight=%22FireSim-generated.sv%22#firesim-buildafi). I'll update the PR to do this since it seems like have 3/4 people willing to remove this.